### DISCUSSION TITLE
Keyboard pops down immediately after popping up, after single tap on input field

### DISCUSSION THREAD

**Discussion by samiul-gtmi** [NativeScript/NativeScript#9657] [Q&A]
/

**Comment by SergeyMell** [comment]
You have a keyboard dismiss in your main view. 
```
<ScrollView row="0" tap="onMainView">
            <StackLayout verticalAlignment="middle" horizontalAlignment="center" class="page-main-content-class"
                         isUserInteractionEnabled="{{ isActivityIndicatorBusy === false}}">
                         ...
```
```
export function onMainView() {
        // hide the keyboard
        console.log('>>>inside mainView')
        dismissKeyboard();
}
```
In android, the event propagates in a way similar to the web one. So in your case when you tap on your input the keyboard shows up, then the event propagates up to the main view, and your function `onMainView` closes the keyboard.

On iOS, such an issue will not appear because iOS events propagate in the opposite way.

**Comment by bradmartin** [comment]
I'd handle the dismissKeyboard differently with your page tap event. You could wire that up on a button tap (submit or login) instead of tapping the page. Also be sure your form uses the return key type for your textfields so the user can move to the textfields via their keyboard control.
